src/coreclr/vm/frozenobjectheap.cpp

```diff
@@ -77,17 +76,33 @@ Object* FrozenObjectHeapManager::TryAllocateObject(PTR_MethodTable type, size_t
 }
 
 
-FrozenObjectSegment::FrozenObjectSegment():
+FrozenObjectSegment::FrozenObjectSegment(size_t size) :
```

Comment:
Maybe call the argument `sizeHint` or something like that since it is not guaranteed to be respected.